src/transformStateWithClones.js

```diff
@@ -6,8 +6,43 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let clonedState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        Object.assign(clonedState, action.extraData);
+
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const key of action.keysToRemove) {
+          delete clonedState[key];
+        }
+
+        break;
+      }
+
+      case 'clear': {
+        clonedState = {};
```

Comment:
You need to figure out how to clear the object differently, since now it is written as a `const`. 
Create a function that will clear the object.
Move the function after `transformStateWithClones` function.